cmake/StaticAnalyzers.cmake

```diff
@@ -62,8 +62,6 @@ macro(enable_clang_format)
             CONFIGURE_DEPENDS # Ensure we re-scan if files change.
             ${PROJECT_SOURCE_DIR}/src/*.[ch]pp
             ${PROJECT_SOURCE_DIR}/src/*.[ch]
-            ${PROJECT_SOURCE_DIR}/tests/*.[ch]pp
-            ${PROJECT_SOURCE_DIR}/tests/*.[ch]
```

Comment:
If we do add C++ examples there then definitely. We seem to be migrating such things out of the repo though so maybe future examples will be in their own project.  

Regardless, in the meantime I think it's good to keep this GLOB as simple as possible though, since it adds to build time (CMake has to re-verify globs every build).